File: 55-JumpGame/55-JumpGame.py
# Last updated: 3/31/2026, 9:40:35 PM
class Solution:
    def canJump(self, nums: List[int]) -> bool:
        maxx = 0
        for i in range(len(nums)):
            if i>maxx:
                return False
            maxx = max(maxx, i+nums[i])
        return True if maxx>=len(nums)-1 else False

File: 55-JumpGame/55-JumpGame.cpp
// Last updated: 3/31/2026, 9:40:42 PM
class Solution {
public:
    bool canJump(vector<int>& nums) {
        int n = nums.size();
        vector<int> freq(n,0);
        freq[0] = 1;
        for(int i=0;i<n-1;i++){
            if (freq[i]==0){
                return false;
            }
            for(int j=i;j<=min(i+nums[i],n-1);j++){
                freq[j]++;
            }
        }
        if (freq[n-1]==0){
            return false;
        }
        else{
            return true;
        }
    }
};